internal/entitlement/connector.go

```diff
@@ -33,68 +33,98 @@ type EntitlementConnector interface {
 }
 
 type entitlementConnector struct {
-	entitlementBalanceConnector EntitlementBalanceConnector
-	entitlementRepo             EntitlementRepo
-	featureConnector            productcatalog.FeatureConnector
+	meteredEntitlementConnector SubTypeConnector
+	staticEntitlementConnector  SubTypeConnector
+	booleanEntitlementConnector SubTypeConnector
+
+	entitlementRepo  EntitlementRepo
+	featureConnector productcatalog.FeatureConnector
 }
 
 func NewEntitlementConnector(
-	entitlementBalanceConnector EntitlementBalanceConnector,
 	entitlementRepo EntitlementRepo,
 	featureConnector productcatalog.FeatureConnector,
-) EntitlementConnector {
+	meteredEntitlementConnector SubTypeConnector,
+	staticEntitlementConnector SubTypeConnector,
+	booleanEntitlementConnector SubTypeConnector,
+) Connector {
 	return &entitlementConnector{
-		entitlementBalanceConnector: entitlementBalanceConnector,
+		meteredEntitlementConnector: meteredEntitlementConnector,
+		staticEntitlementConnector:  staticEntitlementConnector,
+		booleanEntitlementConnector: booleanEntitlementConnector,
 		entitlementRepo:             entitlementRepo,
 		featureConnector:            featureConnector,
 	}
 }
 
 func (c *entitlementConnector) CreateEntitlement(ctx context.Context, input CreateEntitlementInputs) (Entitlement, error) {
-	// TODO: check if the feature exists, if it is compatible with the type, etc....
 	feature, err := c.featureConnector.GetFeature(ctx, models.NamespacedID{Namespace: input.Namespace, ID: input.FeatureID})
 	if err != nil {
 		return Entitlement{}, &productcatalog.FeatureNotFoundError{ID: input.FeatureID}
 	}
-	if feature.ArchivedAt != nil {
+	if feature.ArchivedAt != nil && feature.ArchivedAt.Before(time.Now()) {
```

Comment:
This is a fun use case; you can deprecate/archive a feature in the future.

Reply:
Technically yes, though in practice we always set it for now.